sdk/python/feast/feature_store.py

```diff
@@ -1013,6 +1020,23 @@ def _entity_row_to_field_values(
     return result
 
 
+def _use_time_feature_views(all_feature_views, features):
```

Comment:
Is `use_time` a term people use? It wasn't immediately clear to me what this meant. I was also imagining that the logic for manipulating the feature view would be on the feature view itself, not a separate method. 

Finally, you should add type annotations please.

Reply:
1. Definitely open to alternatives of `use_time` - I agree that it's not clear, but I'm not sure how else to describe it or what to rename it to. Basically instead of grabbing all of the registered `FeatureView`s, we need to grab any changes that the user wants to specify at the time of using the FeatureView (like the `join_key_map` or a new name).

2. By "the logic for manipulating the feature view would be on the feature view itself", do you mean `.override_join_keys` or `.with_join_key_map` should manipulate the original object? I feel like we wouldn't want that behavior with this design given one might use it multiple times in different contexts.

3. Will do in next commit 👌

Reply:
> Definitely open to alternatives of use_time - I agree that it's not clear, but I'm not sure how else to describe it or what to rename it to. Basically instead of grabbing all of the registered FeatureViews, we need to grab any changes that the user wants to specify at the time of using the FeatureView (like the join_key_map or a new name).

I'd first want to see if we can avoid naming it at all. My instinct would have been to use a class like [FeatureViewProjection](https://github.com/feast-dev/feast/blob/master/sdk/python/feast/feature_view_projection.py) which already supports filtering columns, and adding more options like renaming the feature view or overriding the join key. We may change it over time to be `FeatureViewQuery` or something. That would also prevent somebody from doing `fv.with_name("blah")` and then accidentally registering that FV, since we only anticipate the use of renaming to be for queries.

> By "the logic for manipulating the feature view would be on the feature view itself", do you mean .override_join_keys or .with_join_key_map should manipulate the original object? I feel like we wouldn't want that behavior with this design given one might use it multiple times in different contexts.

Not override the original object, but instead produce a new object. It's just that the logic itself seems like it is better to have on the feature view class itself, basically what @mavysavydav did with `with_name()`.